hudi-spark-datasource/hudi-spark/src/main/java/org/apache/hudi/cli/BootstrapExecutorUtils.java

```diff
@@ -269,6 +270,8 @@ private Pair<String, String> genKeyGenClassAndPartitionColumns() {
       keyGenClass = props.getString(HoodieWriteConfig.KEYGENERATOR_CLASS_NAME.key());
     } else if (StringUtils.nonEmpty(props.getString(HoodieWriteConfig.KEYGENERATOR_TYPE.key(), null))) {
       keyGenClass = HoodieSparkKeyGeneratorFactory.getKeyGeneratorClassName(props);
+    } else if (StringUtils.nonEmpty(props.getString(HoodieTableConfig.KEY_GENERATOR_TYPE.key(), null))) {
+      keyGenClass = KeyGeneratorType.valueOf(props.getString(HoodieTableConfig.KEY_GENERATOR_TYPE.key())).getClassName();
```

Comment:
Can the `KeyGeneratorType#getKeyGeneratorClassName` be used instead?

Reply:
done

Reply:
Is this resolved?

Reply:
For this one I left the existing if-else blocks because one of them uses `HoodieSparkKeyGeneratorFactory` which used some infer keygen function. I did not want to change that order here.